kernel/Kconfig

```diff
@@ -862,25 +862,47 @@ config XIP
 
 menu "Security Options"
 
-config STACK_CANARIES
-	bool "Compiler stack canaries"
-	depends on ENTROPY_GENERATOR || TEST_RANDOM_GENERATOR
-	select NEED_LIBC_MEM_PARTITION if !STACK_CANARIES_TLS
+config NEED_STACK_CANARIES
```

Comment:
can we use `REQUIRES_STACK_CANARIES` to stay aligned with other Zephyr toolchain / build system settings, like
`REQUIRES_STD_C99`, `REQUIRES_FULL_LIBC`, etc.

Reply:
Done :)